### ISSUE TITLE
[docs] Best approach to detecting loading and start events

### ISSUE THREAD

**Issue by nigelsim** [Eyevinn/webrtc-player#47]
I'm quite new to WebRTC, so I'm trying to understand the proper way to detect all of the main state transitions, so I can keep the user informed of what is happening.

This is my rough, reverse engineered understanding of the main states, and where the existing events are emitted.

![](https://cdn-0.plantuml.com/plantuml/dpng/ZP71JiCm38RlUGgVaV82bOQ6XX0xW2esmG5n28dBB5e79UwUdwHh5L5b4ZVdz-__gzrj51Da77gbtcy-GD_2dad8z1I9lmxthLgziwL533lFZ5RGNJCy6EhHDV2JQcqlRlAikqbNVp8QU3dkYf9zO4iPnGZ2PfEq1Le8_vtDtDNWNOpZaBuQFj6QCIB82LDQktWhcyrIOIrPT6bKDhqwSaoq3iMUm7SJ-krntqBjFfdyIWxz8V-jr2sF-W4yIp3s6_2CBCKzHptx9tHa6c2lXrofMLbp6fZQEg3rPmm9l4Lsuz3_0W00)

My question is, how should I detect that the video stream is about to start? Should emit events from webrtc-player, or should I be listening to the video element?

Basically, I want to show a loading elements, and bring up messages if things are taking a long time.

Once I've got a good understanding I'm happy to update the main readme file to reflect this.

**Comment by bwallberg** [comment]
Hi @nigelsim!

The most straightforward way to detect anything related to the playback of the video stream itself is to listen to the video element, it will emit events such as `canplaythrough` once it's ready to play and `waiting` if it's buffering ( you can read more about it here: https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement ). 

That being said there are cases, such as startup, where a combination of listening to PeerConnection _and_ the video element would be more suitable. We'd like to implement something like that in the player but unfortunately we don't currently have a timeline for that.
